pkg/handlers/v1/books.go

```diff
@@ -2,56 +2,121 @@ package v1
-	_, parseErr := uuid.Parse(bookId.ID)
-	if parseErr != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": parseErr})
-		return
-	}
-
-	esQuery := query.NewQueryBuilder().ID(bookId.ID).Build()
-	fmt.Println(esQuery)
-	hits, err := clients.DoSearch(c, booksIndex, esQuery)
+	esQuery := query.NewQueryBuilder().ID(bookReq.ID).Build()
+	hits, _, err := clients.DoSearch(c, booksIndex, esQuery, 1, 0)
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"message": err})
+		logrus.Errorf("Error searching for book with ID %s: %v", bookReq.ID, err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
 		return
 	}
 	if len(hits) == 0 {
+		logrus.Infof("Book with ID %s not found", bookReq.ID)
 		c.JSON(http.StatusNotFound, gin.H{"message": "Book not found"})
 		return
 	}
 
-	c.JSON(200, hits[0])
+	logrus.Infof("Book with ID %s retrieved successfully", bookReq.ID)
+	c.JSON(http.StatusOK, hits[0])
 }
 
 func CreateBook(c *gin.Context) {
-	payload, _ := c.Get("validated")
-	bodyBookReq, _ := payload.(req.AddBook)
-	book := common.Book{ID: uuid.New(), PublishDate: bodyBookReq.PublishDate.Format("2006-01-02")}
+	bodyBookReq := utils.GetValidatedPayload[req.AddBook](c)
+	book := common.Book{
+		ID:          uuid.New(),
+		PublishDate: bodyBookReq.PublishDate.Format("2006-01-02"),
+	}
+
+	if err := copier.Copy(&book, &bodyBookReq); err != nil {
+		logrus.Errorf("Error copying AddBook payload to Book: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "Failed to map request data"})
+		return
+	}
+
+	clients.EnqueueIndexTask(c, booksIndex, book.ID.String(), book, _const.CreateIndex)
+	logrus.Infof("Book with ID %s queued for creation successfully", book.ID)
+	c.JSON(http.StatusAccepted, res.AddBook{ID: book.ID})
+}
+
+func UpdateBook(c *gin.Context) {
+	bodyBookReq := utils.GetValidatedPayload[req.UpdateBook](c)
+	titleUpdate := common.TitleUpdate{}
+	if err := copier.Copy(&titleUpdate, &bodyBookReq); err != nil {
+		logrus.Errorf("Error copying UpdateBook payload to TitleUpdate: %v", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": "Failed to map request data"})
+		return
+	}
+
+	clients.EnqueueIndexTask(c, booksIndex, bodyBookReq.ID, titleUpdate, _const.UpdateIndex)
+	logrus.Infof("Book with ID %s queued for update successfully", bodyBookReq.ID)
+	c.JSON(http.StatusAccepted, res.UpdateBook{ID: uuid.MustParse(bodyBookReq.ID)})
+}
+
+func DeleteBook(c *gin.Context) {
+	deleteReq := utils.GetValidatedPayload[req.DeleteBook](c)
 
-	_ = copier.Copy(&book, &bodyBookReq)
+	clients.EnqueueIndexTask(c, booksIndex, deleteReq.ID, "", _const.DeleteIndex)
+	logrus.Infof("Book with ID %s queued for deletion successfully", deleteReq.ID)
+	c.JSON(http.StatusAccepted, res.DeleteBook{ID: uuid.MustParse(deleteReq.ID)})
+}
+
+func SearchBooks(c *gin.Context) {
+	searchReq := utils.GetValidatedPayload[req.SearchBooks](c)
+
+	esQuery := query.NewQueryBuilder().
```

Comment:
As the handler shouldn't know it deals with Elastic, the variable should just be `query`